app/models/file_depot_s3.rb

```diff
@@ -0,0 +1,95 @@
+class FileDepotS3 < FileDepot
+  attr_accessor :s3
+
+  def self.uri_prefix
+    "s3"
+  end
+
+  def self.validate_settings(settings)
+    _log.debug("Validating settings [#{settings}]")
+    new(:uri => settings[:uri]).verify_credentials(nil, settings.slice(:username, :password))
+  end
+
+  extend ActiveSupport::Concern
+
+  #
+  # Connections
+  #
+  def connect(options = {})
+    username = options[:username] || authentication_userid(options[:auth_type])
+    password = options[:password] || authentication_password(options[:auth_type])
+    service  = :S3
```

Comment:
If we are hard-coding the service, can we use `Aws::S3::Resource.new` rather than passing this around and doing a `const_get`?